internal/auth/providers/okta.go

```diff
@@ -0,0 +1,416 @@
+package providers
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/buzzfeed/sso/internal/auth/circuit"
+	"github.com/buzzfeed/sso/internal/pkg/groups"
+	log "github.com/buzzfeed/sso/internal/pkg/logging"
+	"github.com/buzzfeed/sso/internal/pkg/sessions"
+	"github.com/datadog/datadog-go/statsd"
+)
+
+// OktaProvider is an implementation of the Provider interface.
+type OktaProvider struct {
+	*ProviderData
+	StatsdClient *statsd.Client
+	cb           *circuit.Breaker
+	GroupsCache  groups.MemberSetCache
+}
+
+type UserInfoResponse struct {
```

Comment:
Let's make this an inline anonymous struct, which appears to be stylistically more consistent with other parts of this package.

Reply:
Clarified that in this case `UserInfoResponse` is passed around and used in multiple functions, which is why it was created as a global struct like so. Possible the logic that uses it could be altered to better match other examples in the package but for now I think it's alright to leave it like so?